layers/+config-files/puppet/packages.el

```diff
@@ -38,4 +38,4 @@
   (spacemacs|add-company-hook puppet-mode))
 
 (defun puppet/post-init-flycheck ()
-  (spacemacs/add-flycheck-hook 'puppet-mode-hook))
+  (push 'puppet-mode flycheck-global-modes))
```

Comment:
I would keep a function for this purpose, because it's a little more complicated than `push`.  It's more like

```
(when (and (listp flycheck-global-modes) (not (eq 'not (car flycheck-global-modes))) 
   (add-to-list 'flycheck-global-modes mode))
```

to account for `flycheck-global-modes` set to `t` (first case) or `(not ...)` (second case) by the user.